src/sys/db/TableCreate.hx

```diff
@@ -86,13 +86,28 @@ class TableCreate {
 			}
 			decls.push(quote(f.name)+" "+getTypeSQL(f.t,dbName)+(f.isNull ? "" : " NOT NULL"));
 		}
-		if( dbName != "SQLite" || !hasID )
```

Comment:
This previous version of the line was likely the correct one. 
If we get right of the `dbName != "SQLite"` predicate, then for MySQL the PRIMARY KEY declaration will *only* happen when type has no DID, DUId, DBigId fields. 

Instead we want it all the time for MySQL.
This should solve the unit tests that failed on mysql.